### ISSUE TITLE
False positive in python.lang.correctness.useless-eqeq.useless-eqeq

### ISSUE THREAD

**Issue by hdkire** [returntocorp/sgrep-rules#1470]
**Describe the bug**

`v != v` and  `v == v` are useful with numpy when testing for NAN and NAT. 

It is reported as a useless comparison:

```
This expression is always True: `v == v` or `v != v`. If testing for floating point NaN, use `math.isnan(v)`, or `cmath.isnan(v)` if the number is complex
```
The suggested fix does not work for NAT, or other possible values that do not equal themselves.  Also raises TypeError for strings.

**To Reproduce**
```python
import numpy as np


def nan_or_nat(v):
    return v != v


print(nan_or_nat(np.nan))
print(nan_or_nat(np.datetime64('nat')))
print(nan_or_nat(42))
print(nan_or_nat('test'))
```
It would not be convenient to implement using math.isnan.

**Expected behavior**
Do not mark `v != v` as a problem

**Priority**
How important is this to you?
- [ ] P0: blocking me from making progress
- [ ] P1: this will block me in the near future
- [x] P2: annoying but not blocking me

**Comment by minusworld** [comment]
Hey @hdkire! 👋  Thanks for filing an issue.

It's definitely not ideal for Semgrep to report this finding. However, the rule is tailored toward the general case, and Semgrep is unable to reason about the specific context (such as types) in order to automatically make a judgement.

Are you able to use a `# nosemgrep` comment in order to suppress the finding? This is probably the easiest solution since you will know the code best. Let us know -- thanks!

**Comment by hdkire** [comment]
Alright, thanks for the answer. I will go with the easy solution.
